src/Microsoft.AspNetCore.Authentication.Core/AuthenticationService.cs

```diff
@@ -89,14 +90,17 @@ public AuthenticationService(IAuthenticationSchemeProvider schemes, IAuthenticat
                 scheme = defaultChallengeScheme?.Name;
                 if (scheme == null)
                 {
-                    throw new InvalidOperationException($"No authenticationScheme was specified, and there was no DefaultChallengeScheme found.");
+                    context.Response.StatusCode = StatusCodes.Status401Unauthorized;
```

Comment:
Note that returning a 401 response without a `WWW-Authenticate` header is illegal:

> A server generating a 401 (Unauthorized) response MUST send a WWW-Authenticate header field containing at least one challenge. A server MAY generate a WWW-Authenticate header field in other response messages to indicate that supplying credentials (or different credentials) might affect the response.

https://tools.ietf.org/html/rfc7235#section-4.1